### PR TITLE
chore: split mgmt/gen.go into several files

### PR DESCRIPTION
**Pull request by AlekSi**
No functional changes in this PR, to make future PRs easier.

### CODE REVIEW

File: cmd/talosctl/cmd/mgmt/gen/config.go [hunk 1]
```diff
@@ -110,10 +110,10 @@ func fixControlPlaneEndpoint(u *url.URL) *url.URL {
 	return u
 }
 
-// GenV1Alpha1Config generates the Talos config bundle
+// genV1Alpha1Config generates the Talos config bundle
 //
-// GenV1Alpha1Config is useful for integration with external tooling options.
-func GenV1Alpha1Config(genOptions []generate.GenOption,
```
Comment (smira) [inline cmd/talosctl/cmd/mgmt/gen/config.go:116]: This function was made public on purpose in one of the previous PRs. Even if that is not the perfect place for it, let's not break the code which imports it

File: pkg/machinery/config/types/v1alpha1/generate/generate.go [hunk 2]
```diff
@@ -381,8 +381,8 @@ func NewTalosCA(currentTime time.Time) (ca *x509.CertificateAuthority, err error
 	return x509.NewSelfSignedCertificateAuthority(opts...)
 }
 
-// NewAdminCertificateAndKey generates the admin Talos certificate and key.
-func NewAdminCertificateAndKey(currentTime time.Time, ca *x509.PEMEncodedCertificateAndKey, loopback string) (p *x509.PEMEncodedCertificateAndKey, err error) {
+// newAdminCertificateAndKey generates the admin Talos certificate and key.
+func newAdminCertificateAndKey(currentTime time.Time, ca *x509.PEMEncodedCertificateAndKey, loopback string) (p *x509.PEMEncodedCertificateAndKey, err error) {
```
Comment (smira) [inline pkg/machinery/config/types/v1alpha1/generate/generate.go:385]: Same here, these functions are public interface, and some of them are used in CAPI providers

File: cmd/talosctl/cmd/mgmt/gen/config.go (hunk 1 above)
Comment (AlekSi) [reply cmd/talosctl/cmd/mgmt/gen/config.go:116]: https://github.com/talos-systems/talos/pull/3650

**Review by smira — APPROVED** [review]

**Pull request merged**